pyomo/contrib/parmest/parmest.py

```diff
@@ -848,99 +910,117 @@ def leaveNout_bootstrap_test(self, lNo, lNo_samples, bootstrap_samples,
         assert distribution in ['Rect', 'MVN', 'KDE']
         assert isinstance(alphas, list)
         assert isinstance(seed, (type(None), int))
-        
+
         if seed is not None:
             np.random.seed(seed)
-            
+
         data = self.callback_data.copy()
-        
+
         global_list = self._get_sample_list(lNo, lNo_samples, replacement=False)
-            
+
         results = []
         for idx, sample in global_list:
-            
+
             # Reset callback_data to only include the sample
             self.callback_data = [data[i] for i in sample]
 
             obj, theta = self.theta_est()
-            
+
             # Reset callback_data to include all scenarios except the sample
             self.callback_data = [data[i] for i in range(len(data)) if i not in sample]
 
             bootstrap_theta = self.theta_est_bootstrap(bootstrap_samples)
-            
-            training, test = self.confidence_region_test(bootstrap_theta, 
-                                    distribution=distribution, alphas=alphas, 
+
+            training, test = self.confidence_region_test(bootstrap_theta,
+                                    distribution=distribution, alphas=alphas,
                                     test_theta_values=theta)
-                
+
             results.append((sample, test, training))
-        
+
         # Reset callback_data (back to full data set)
         self.callback_data = data
-        
+
         return results
-    
-    
-    def objective_at_theta(self, theta_values):
+
+
+    def objective_at_theta(self, theta_values=None,
+                            initialize_parmest_model = False):
         """
         Objective value for each theta
 
         Parameters
         ----------
         theta_values: pd.DataFrame, columns=theta_names
             Values of theta used to compute the objective
-            
+
+        initialize_parmest_model: boolean
+            True: Solve square problem instance, build extensive form of the model for 
+            parameter estimation, and set flag model_initialized to True
+
+
         Returns
         -------
         obj_at_theta: pd.DataFrame
-            Objective value for each theta (infeasible solutions are 
+            Objective value for each theta (infeasible solutions are
             omitted).
         """
-        assert isinstance(theta_values, pd.DataFrame)
-        
-        # for parallel code we need to use lists and dicts in the loop
-        theta_names = theta_values.columns
-        all_thetas = theta_values.to_dict('records')
+        if theta_values is None:
+            all_thetas = {} # dictionary to store fitted variables
+            theta_names = self.theta_names
+        else:
+            assert isinstance(theta_values, pd.DataFrame)
+
+            # for parallel code we need to use lists and dicts in the loop
+            theta_names = theta_values.columns
```

Comment:
Is it ok if `theta_names` does not equal `self.theta_names`?

Reply:
Given the way parmest modifies the model, I think these two list of names needs to match.  A check would be good.

Reply:
I added a check for `theta_names` to the equal to `self.theta_names` only when `initialize_parmest_model=True`. I am not sure how to check this for the different model variants being tested in lines 291 - 418 of https://github.com/Pyomo/pyomo/blob/main/pyomo/contrib/parmest/tests/test_parmest.py

Reply:
@kaklise I ran into a few issues while writing the check to compare `theta_names` and `self.theta_names`: the check fails for certain model variants specified in the test in lines 291 - 412 of https://github.com/Pyomo/pyomo/blob/main/pyomo/contrib/parmest/tests/test_parmest.py: 

1. Check fails for model variants `param_index` and `vars_index` since `theta_names` and column names of `theta_vals` are different
2. Check fails for model variants `vars_quoted_index` and `vars_str_index`: `theta_est()` modifies `self.theta_names` in line 368 of https://github.com/Pyomo/pyomo/blob/main/pyomo/contrib/parmest/parmest.py and removes quotes from indexed parameter names (e.g. `theta['asymptote']` becomes `theta[asymptote]`)

I am working on fixing these check fails.  Please let me know if you have any suggestions/comments